demo/full/scripts/modules/player/index.js

```diff
@@ -127,16 +126,33 @@ const PLAYER = ({ $destroy, state }, { videoElement, textTrackElement }) => {
       player.setVolume(volume);
     },
 
-    LOAD: (arg) => {
+    LOAD: (arg, nextInitOptions) => {
       dettachVideoThumbnailLoader();
+      if (JSON.stringify(initOptions) !== JSON.stringify(nextInitOptions)) {
```

Comment:
We're comparing the aspect of the JSON string once serialized from a JS object.
That's not a beautiful solution because it might depend on key ordering and it kind of seems as wasteful.

I'm sure we can have a better solution without much effort.

Reply:
Also it doesn't work with non-serializable objects like ArrayBuffers and functions.

The better solution would be to just "know" which values have been changed but I'm unsure if this is easily doable.